tests/api/runtimes/base.py

```diff
@@ -157,10 +158,16 @@ def _assert_function_config(
     @staticmethod
     def _assert_pod_env(pod_env, expected_variables):
         for env_variable in pod_env:
-            name = env_variable["name"]
+            if isinstance(env_variable, V1EnvVar):
```

Comment:
You're doing the `isinstance` check in multiple places. It's better to convert `env_variable` in a single place:
```
if isinstance(env_variable, dict):
    env_variable = V1EnvVar(name=env_variable["name"], value=env_variable["value"])
```
And then you can just use the `V1EnvVar` accessors.